src/Common/Core/Impl/Extensions/CultureExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.Common.Core {
+    public static class CultureExtensions {
+        private static readonly char[] _separators = new char[] { ',', ' ' };
+
+        public static string LanguageNameFromLCID(int lcid) {
+            var culture = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures).FirstOrDefault(x => x.LCID == lcid);
+            return culture.GetLanguageName();
+        }
+
+        public static IEnumerable<string> GetLanguageNames() {
+            IEnumerable<CultureInfo> cultures = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures);
+            return cultures.Select(c => c.GetLanguageName()).Distinct().OrderBy(x => x);
```

Comment:
Are we sure that for the cases where there will be duplicates (I'm assuming this is after stripping things after separators), after eliminating them, the remaining entry will be the one that the user wants?

E.g. for Chinese, there are two common encodings, Big5 and GB. If we show only one entry for "China", and always pick one and never the other, it will present a problem.